spec.html

```diff
@@ -39139,13 +38971,12 @@ <h1>
       <emu-clause id="sec-isdetachedbuffer" type="abstract operation">
         <h1>
           IsDetachedBuffer (
-            _arrayBuffer_: unknown,
+            _arrayBuffer_: an Object that has an [[ArrayBufferData]] internal slot,
```

Comment:
```suggestion
            _arrayBuffer_: an ArrayBuffer or a SharedArrayBuffer,
```

Reply:
Do these have definitions at all?

Reply:
If we're being precise, "an ArrayBuffer" means "a value which was created by AllocateArrayBuffer", and "a SharedArrayBuffer" means "a value which was created by AllocateSharedArrayBuffer". (Similarly, "a TypedArray" means "a value which was created by AllocateTypedArray", "an Array" means "a value which was created by ArrayCreate", etc.)

Equivalently, ArrayBuffers are those objects with [[ArrayBufferData]], [[ArrayBufferByteLength]], and [[ArrayBufferDetachKey]] slots, and SharedArrayBuffers are those objects with [[ArrayBufferData]] and [[ArrayBufferByteLength]] slots but not a [[ArrayBufferDetachKey]] slot.

Reply:
Yes, but where in the spec is that defined explicitly? Do we use “a TypedArray” and similar anywhere else?

Reply:
> Yes, but where in the spec is that defined explicitly?

[AllocateArrayBuffer](https://tc39.es/ecma262/multipage/structured-data.html#sec-allocatearraybuffer) says "It is used to create an ArrayBuffer object".

> Do we use “a TypedArray” and similar anywhere else?

There are number of existing uses of "a TypedArray object" and "an ArrayBuffer object" as parameter types, yes.

Reply:
ok, so we're using the phrasing "a(n) X object" in this PR, then?

Reply:
> ok, so we're using the phrasing "a(n) X object" in this PR, then?

You can see for yourself that it uses "an X" (including changing existing usages). I agree that that change is basically orthogonal, but I don't mind including it here.

Reply:
It seems like it would be useful to make that change in a separate PR - in either order. Lots of people look at changes to the spec PR by PR, so readability/understandability isn't solely determined by the final output - and this is already a massive PR.